tests/python/unittest/test_gluon_event_handler.py

```diff
@@ -0,0 +1,92 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import tempfile
+import mxnet as mx
+from mxnet import nd
+from mxnet.gluon import nn, loss
+from mxnet.gluon.estimator import estimator, event_handler
+
+def _get_test_network():
+    net = nn.Sequential()
+    net.add(nn.Dense(128, activation='relu', in_units=100, flatten=False),
+              nn.Dense(64, activation='relu', in_units=128),
+              nn.Dense(10, activation='relu', in_units=64))
+    return net
+
+def _get_test_data():
+    return mx.io.NDArrayIter(data=nd.ones((32, 100)), label=nd.random.randint(0, 10, (32, 1)))
+
+
+def test_checkpoint_handler():
+    tmpdir = tempfile.mkdtemp()
+    file_path = os.path.join(tmpdir, "model.params")
+    test_data  = _get_test_data()
+
+    save_best_only = False
+    mode = 'auto'
+
+    net = _get_test_network()
+    ce_loss = loss.SoftmaxCrossEntropyLoss()
+    acc = mx.metric.Accuracy()
+    est = estimator.Estimator(net, loss=ce_loss, metrics=acc)
+    checkpoint_handler = [event_handler.CheckpointHandler(est, file_path,
+                                                          save_best_only=save_best_only,
+                                                          mode=mode)]
+    est.fit(test_data, event_handlers=checkpoint_handler, epochs=1)
+    assert os.path.isfile(file_path)
+    os.remove(file_path)
+
+def test_early_stopping():
+    test_data = _get_test_data()
+
+    mode = 'max'
+    monitor = 'train_accuracy'
+    patience = 0
+
+    net = _get_test_network()
+    ce_loss = loss.SoftmaxCrossEntropyLoss()
+    acc = mx.metric.Accuracy()
+    est = estimator.Estimator(net, loss=ce_loss, metrics=acc)
+    early_stopping = [event_handler.EarlyStoppingHandler(est, monitor,
+                                                         patience=patience,
+                                                          mode=mode)]
+    est.fit(test_data, event_handlers=early_stopping, epochs=1)
+
+    mode = 'auto'
+    monitor = 'train_accuracy'
+    patience = 2
+    early_stopping = [event_handler.EarlyStoppingHandler(est, monitor,
+                                                         patience=patience,
+                                                          mode=mode)]
+    est.fit(test_data, event_handlers=early_stopping, epochs=1)
+
+def test_logging():
+    tmpdir = tempfile.mkdtemp()
+    test_data = _get_test_data()
+    file_name = 'test_log'
+    output_dir = os.path.join(tmpdir, file_name)
+
+    net = _get_test_network()
+    ce_loss = loss.SoftmaxCrossEntropyLoss()
+    acc = mx.metric.Accuracy()
+    est = estimator.Estimator(net, loss=ce_loss, metrics=acc)
+    logging_handler = [event_handler.LoggingHandler(est, file_name=file_name, file_location=tmpdir)]
+    est.fit(test_data, event_handlers=logging_handler, epochs=1)
+    assert os.path.isfile(output_dir)
```

Comment:
do you want to do a sanity check that validation accuracy for last epoch is logged(also that training process completed?